python/ray/data/preprocessor.py

```diff
@@ -245,19 +245,7 @@ def _transform(self, dataset: Dataset) -> Dataset:
                 "and 'arrow' Dataset formats are supported."
             )
 
-        transform_type = self._determine_transform_to_use(dataset_format)
-
-        # Our user facing batch format should only be pandas or numpy, other
-        # formats {arrow, simple} are internal.
-        if transform_type == "pandas":
-            return dataset.map_batches(self._transform_pandas, batch_format="pandas")
-        elif transform_type == "numpy":
-            return dataset.map_batches(self._transform_numpy, batch_format="numpy")
-        else:
-            raise ValueError(
-                "Invalid transform type returned from _determine_transform_to_use; "
-                f'"pandas" and "numpy" allowed, but got: {transform_type}'
-            )
+        return dataset.map_batches(self._transform_batch, batch_format=dataset_format)
```

Comment:
dataset has `arrow` format, but we don't have equivalent transform type / batch_format right ? Basically all data formats in this error message from CI  

```
ValueError: The batch format must be one of ['default', 'native', 'pandas', 'pyarrow', 'numpy'], got: arrow
```

needs to map to either `pandas` or `numpy` batch_format

Reply:
ah yeah dataset format is arrow, but batch format should be pyarrow...

Reply:
> 

should we ensure these two are just the same ? I don't see a reason why a python codebase that don't use pyarrow to handle arrow format ..